File: 백준/Silver/1347. 미로 만들기/미로 만들기.py
dirX = [1,0,-1,0]    #하 좌 상 우 (+ 우회전)
dirY = [0,-1,0,1]

import sys
input = sys.stdin.readline

N = int(input())

move = input()
dir = 0

x = 0
y = 0

minX = 0
minY = 0
maxX = 0
maxY = 0

saveGround = [[0, 0]]

for i in range(N) :
    if move[i] == 'R' :
        dir = (dir + 1) % 4
    elif move[i] == 'L' :
        dir = (dir - 1) % 4
    else :
        x += dirX[dir]
        y += dirY[dir]
        minX = min(minX, x)
        minY = min(minY, y)
        maxX = max(maxX, x)
        maxY = max(maxY, y)
        saveGround.append([x, y])

ground = [['#' for _ in range(maxY - minY + 1)]for _ in range(maxX - minX+ 1)]

while len(saveGround) != 0 :
    pop = saveGround.pop()
    ground[pop[0] - minX][pop[1] - minY] = '.'

for i in range(maxX - minX + 1) :
    for j in range(maxY - minY + 1) :
        print(ground[i][j], end="")
    print()

File: 백준/Silver/1347. 미로 만들기/미로 만들기.java
import java.io.*;
import java.util.*;

public class Main {
	static final int[] dirX = {1,0,-1,0};	//하좌우상
	static final int[] dirY = {0,-1,0,1};
	
	public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st;
        
        st = new StringTokenizer(br.readLine());
    	int N = Integer.parseInt(st.nextToken());
    	
    	int x = 0, y = 0;
    	int dir = 0;
    	int minX = 0, maxX = 0, minY = 0, maxY = 0;
    	
    	char[] move = br.readLine().toCharArray();
    	
    	ArrayDeque<int[]> q = new ArrayDeque<>();
    	
    	q.offer(new int[] {0,0});
    	
    	for (int i = 0; i < N; i++) {
			if (move[i] == 'R') {
				dir = (dir + 1) % 4;
			} else if (move[i] == 'L') {
				dir = (dir + 3) % 4;
			} else {
				x += dirX[dir];
				y += dirY[dir];

				minX = Math.min(minX, x);
				minY = Math.min(minY, y);
				maxX = Math.max(maxX, x);
				maxY = Math.max(maxY, y);
				
				q.offer(new int[] {x, y});
			}
		}
    	
    	int lenX = maxX - minX + 1;
    	int lenY = maxY - minY + 1;
		int[][] map = new int[lenX][lenY];
		
		while (!q.isEmpty()) {
			int[] poll = q.poll();
			
			map[poll[0] - minX][poll[1] - minY] = 1;
		}
		
		for (int i = 0; i < lenX; i++) {
			for (int j = 0; j < lenY; j++) {
				sb.append(map[i][j]==0?'#':'.');
			}
			sb.append("\n");
		}
    	
        System.out.println(sb);
        br.close();
    }
}